diff --git a/datamodules/base/datamodule.py b/datamodules/base/datamodule.py
index 56453a7..ee9b9c9 100644
--- a/datamodules/base/datamodule.py
+++ b/datamodules/base/datamodule.py
@@ -4,8 +4,6 @@ from pathlib import Path
 
 import albumentations as A
 from albumentations.pytorch import ToTensorV2
-from anomalib.data.base.datamodule import collate_fn
-from anomalib.data.utils import InputNormalizationMethod
 from pytorch_lightning import LightningDataModule
 from pytorch_lightning.utilities.types import TRAIN_DATALOADERS, EVAL_DATALOADERS
 from torch.utils.data import DataLoader
@@ -13,6 +11,10 @@ from torch.utils.data import DataLoader
 from datamodules.base import Supervision
 from datamodules.base.dataset import SSNDataset
 
+from enum import Enum
+class InputNormalizationMethod(str, Enum):
+    IMAGENET = "imagenet"
+    NONE = "none"
 
 class SSNDataModule(LightningDataModule, ABC):
     """
@@ -82,11 +84,13 @@ class SSNDataModule(LightningDataModule, ABC):
         self.transform_train = A.Compose(
             base_transforms,
             additional_targets={"loss_mask": "mask"},
+            is_check_shapes=False,
         )
 
         self.transform_eval = A.Compose(
             base_transforms,
             additional_targets={"loss_mask": "mask"},
+            is_check_shapes=False,
         )
 
     @property
@@ -151,5 +155,4 @@ class SSNDataModule(LightningDataModule, ABC):
             shuffle=False,
             batch_size=self.eval_batch_size,
             num_workers=self.num_workers,
-            collate_fn=collate_fn,
         )
diff --git a/datamodules/base/dataset.py b/datamodules/base/dataset.py
index d5e80fd..534a5c5 100644
--- a/datamodules/base/dataset.py
+++ b/datamodules/base/dataset.py
@@ -197,16 +197,14 @@ class SSNDataset(Dataset):
                 image_path = self._normal_samples.iloc[ix].image_path
                 mask_path = self._normal_samples.iloc[ix].mask_path
                 label_index = self._normal_samples.iloc[ix].label_index
-                is_segmented = self._normal_samples.iloc[ix].get("is_segmented", True)
+                is_segmented = bool(self._normal_samples.iloc[ix].get("is_segmented", True))
             else:
                 # to get actual index of positive
                 ix = index % self.num_pos
                 image_path = self._anomalous_samples.iloc[ix].image_path
                 mask_path = self._anomalous_samples.iloc[ix].mask_path
                 label_index = self._anomalous_samples.iloc[ix].label_index
-                is_segmented = self._anomalous_samples.iloc[ix].get(
-                    "is_segmented", True
-                )
+                is_segmented = bool(self._anomalous_samples.iloc[ix].get("is_segmented", True))
         # test
         else:
             if index < self.num_neg:
@@ -214,16 +212,14 @@ class SSNDataset(Dataset):
                 image_path = self._normal_samples.iloc[ix].image_path
                 mask_path = self._normal_samples.iloc[ix].mask_path
                 label_index = self._normal_samples.iloc[ix].label_index
-                is_segmented = self._normal_samples.iloc[ix].get("is_segmented", True)
+                is_segmented = bool(self._normal_samples.iloc[ix].get("is_segmented", True))
 
             else:
                 ix = index - self.num_neg
                 image_path = self._anomalous_samples.iloc[ix].image_path
                 mask_path = self._anomalous_samples.iloc[ix].mask_path
                 label_index = self._anomalous_samples.iloc[ix].label_index
-                is_segmented = self._anomalous_samples.iloc[ix].get(
-                    "is_segmented", True
-                )
+                is_segmented = bool(self._anomalous_samples.iloc[ix].get("is_segmented", True))
 
         return image_path, mask_path, label_index, is_segmented
 
@@ -317,7 +313,8 @@ class SSNDataset(Dataset):
 
         image_path, mask_path, label_index, is_segmented = self.get_sample_data(index)
 
-        image = read_image(image_path)
+        image = cv2.imread(image_path)
+        image = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
         item = dict(
             image_path=image_path,
             label=label_index,
diff --git a/model/feature_extractor.py b/model/feature_extractor.py
index 4ee752e..28531cb 100644
--- a/model/feature_extractor.py
+++ b/model/feature_extractor.py
@@ -1,6 +1,7 @@
 import torch
 import torch.nn as nn
 import torch.nn.functional as F
+import os
 
 from torch import Tensor
 from torchvision.models.feature_extraction import create_feature_extractor
@@ -12,6 +13,7 @@ class FeatureExtractor(nn.Module):
     def __init__(
         self,
         backbone: str,
+        weights: str,
         layers: list[str],
         patch_size: int,
         image_size: tuple[int, int],
@@ -24,6 +26,11 @@ class FeatureExtractor(nn.Module):
             models = import_module("torchvision.models")
             backbone_class = getattr(models, backbone)
             model = backbone_class(weights="IMAGENET1K_V1")
+            if os.path.exists(weights):
+                model = backbone_class()
+                model.load_state_dict(torch.load(weights))
+            else:
+                model = backbone_class(weights=weights)
         except AttributeError as e:
             print(f"Backbone {backbone} not found in torchvision.models.")
             raise AttributeError from e
diff --git a/model/supersimplenet.py b/model/supersimplenet.py
index 40b71a8..26e8810 100644
--- a/model/supersimplenet.py
+++ b/model/supersimplenet.py
@@ -29,6 +29,7 @@ class SuperSimpleNet(nn.Module):
         self.config = config
         self.feature_extractor = FeatureExtractor(
             backbone=config.get("backbone", "wide_resnet50_2"),
+            weights=config.get("backbone_weights"),
             layers=config.get("layers", ["layer2", "layer3"]),
             patch_size=config.get("patch_size", 3),
             image_size=image_size,
