diff --git a/common/visualizer.py b/common/visualizer.py
index 9682500..54c32cd 100644
--- a/common/visualizer.py
+++ b/common/visualizer.py
@@ -2,10 +2,11 @@ from pathlib import Path
 
 import cv2
 import numpy as np
-from anomalib.data.utils import read_image
+from PIL import Image, ImageOps
 from matplotlib import pyplot as plt
 
 from tqdm import tqdm
+import mlflow
 
 
 class Visualizer:
@@ -15,21 +16,24 @@ class Visualizer:
     def visualize(self, results: dict):
         for image_path, mask_path, anomaly_map, score, seg_score, label in tqdm(
             zip(
-                results["image_path"],
-                results["mask_path"],
-                results["anomaly_map"],
-                results["score"],
-                results["seg_score"],
-                results["label"],
+                results.image_path,
+                results.mask_path,
+                results.anomaly_map,
+                results.score,
+                results.seg_score,
+                results.label,
             ),
-            total=len(results["label"]),
+            total=len(results.label),
+            desc="visualise",
         ):
             anomaly_map = anomaly_map.squeeze()
             h, w = anomaly_map.shape
-            image = read_image(image_path, image_size=(h, w))
+            image = cv2.imread(image_path)
+            image = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
             # if no mask path -> all zeros
             if mask_path:
-                gt_mask = read_image(mask_path, image_size=(h, w)).squeeze()
+                gt_mask = cv2.imread(mask_path)
+                gt_mask = cv2.cvtColor(gt_mask, cv2.COLOR_BGR2RGB)
             else:
                 gt_mask = np.zeros_like(anomaly_map)
             # images are normed to [0, 1] so we cut off at 0.5
@@ -38,14 +42,14 @@ class Visualizer:
             gt_label = "Anomalous" if label.item() else "Normal"
 
             # adjust height and width as 256 as baseline
-            fig_h = h / 256 * 2
+            fig_h = h / 256 * 4
             fig_w = w / 256 * 12
 
             fig, plots = plt.subplots(1, 5, figsize=(fig_w, fig_h))
             for s_plt in plots:
                 s_plt.axis("off")
 
-            fig.tight_layout()
+            fig.tight_layout(rect=[0,0,1,0.9])
 
             plots[0].imshow(image)
             plots[0].title.set_text("Image")
@@ -66,17 +70,13 @@ class Visualizer:
 
             # subdir name is parent's name
             subdir_name = Path(image_path).parent.name
-            plot_name = f"{Path(image_path).stem}.png"
+            plot_name = f"{subdir_name}_{Path(image_path).stem}.png"
 
             # besides parent name also add anomalous or normal
-            total_path = self.save_path / subdir_name / gt_label
+            total_path = self.save_path / gt_label #  / subdir_name
             total_path.mkdir(exist_ok=True, parents=True)
 
-            plt.savefig(total_path / plot_name, bbox_inches="tight")
-
-            ano_maps_dir = total_path / "anomaly_maps"
-            ano_maps_dir.mkdir(exist_ok=True, parents=True)
-
-            cv2.imwrite(str(ano_maps_dir / plot_name), anomaly_map.numpy() * 255)
+            mlflow.log_figure(fig, total_path / plot_name)
+            # mlflow.log_image((anomaly_map.numpy() * 255).astype(int), total_path / "anomaly_maps" / plot_name)
 
             plt.close("all")
